bundle/parameters.go

```diff
@@ -23,10 +28,30 @@ func (p *Parameter) AppliesTo(action string) bool {
 }
 
 // Validate a Parameter
-func (p *Parameter) Validate() error {
+func (p *Parameter) Validate(name string, bun Bundle) error {
 	if p.Definition == "" {
 		return errors.New("parameter definition must be provided")
 	}
+
+	// Validate default against definition schema, if exists
+	schema, ok := bun.Definitions[p.Definition]
+	if !ok {
+		return fmt.Errorf("unable to find definition for %s", name)
+	}
+	if schema.Default != nil {
+		var valResult *multierror.Error
+		valErrs, err := schema.Validate(schema.Default)
+		if err != nil {
+			valResult = multierror.Append(valResult, errors.Wrapf(err, "encountered an error validating parameter %s", name))
```

Comment:
Let's sneak something about this being a default value for the parameter being validated. So that it's clear exactly which part of the parameter is messed up.

```suggestion
			valResult = multierror.Append(valResult, errors.Wrapf(err, "encountered an error validating the default value %v for parameter %s", schema.Default, name))
```

Reply:
Oops, haha, had thought of this and pushed a commit without checking in on this review.  Good call.  I'll incorporate the full suggestion you've added.